python-libraries/nanover-core/src/nanover/imd/imd_force.py

```diff
@@ -8,13 +8,23 @@
 """
 
 from math import exp
-from typing import Tuple, Optional, Iterable
+from typing import Tuple, Optional, Iterable, Dict, Protocol
 
 import numpy as np
 import numpy.typing as npt
 from nanover.imd.particle_interaction import ParticleInteraction
 
 
+class ForceCalculator(Protocol):
+    def __call__(
+        self,
+        positions: npt.NDArray,
+        interaction_position: npt.NDArray,
+        *,
+        periodic_box_lengths: Optional[npt.NDArray],
+    ) -> Tuple[float, npt.NDArray]: ...
```

Comment:
Maybe I have to change this to `**kwargs: Unpack[...]` or something to get what we really want, but either way this whole thing is exposing that some of the force calculators have parameters that are never varied in the server and aren't covered by tests as far as I can see (k and sigma).

Reply:
I think we can put `k` and `sigma` in the function body. We do not plan on exposing these variables.